compiler/src/dotty/tools/dotc/transform/Splicer.scala

```diff
@@ -313,9 +313,14 @@ object Splicer {
       case Call(fn, args) =>
         if (fn.symbol.isConstructor && fn.symbol.owner.owner.is(Package)) {
           interpretNew(fn.symbol, args.map(interpretTree))
+        } else if (fn.symbol.is(Module)) {
+          interpretModuleAccess(fn.symbol)
         } else if (fn.symbol.isStatic) {
-          if (fn.symbol.is(Module)) interpretModuleAccess(fn.symbol)
-          else interpretStaticMethodCall(fn.symbol, args.map(arg => interpretTree(arg)))
+          val module = fn.symbol.owner
+          interpretStaticMethodCall(module, fn.symbol, args.map(arg => interpretTree(arg)))
+        } else if (fn.qualifier.symbol.is(Module)) {
+          val module = fn.qualifier.symbol.moduleClass
+          interpretStaticMethodCall(module, fn.symbol, args.map(arg => interpretTree(arg)))
```

Comment:
How we can be sure that `module` is static?

Reply:
I should add that condition

Reply:
Added